### PR TITLE
GH-8361: Fixed editor resizing issue in Output.

### LINKED ISSUE #8361: [output] The monaco editor does not resize correctly when other views are maximized
### Steps to Reproduce:

1. Check the screencast.

screencast 2020-08-12 13-22-37

### Additional Information

- Operating System:
- Theia Version:

### PR DESCRIPTION
**Pull request by kittaakos**
Closes #8361.

Signed-off-by: Akos Kitta <kittaakos@typefox.io>

#### What it does

Fixes the editor resizing issue in the Output view.

#### How to test

 - Make sure the `API sample` Output channel is visible.
 - Maximize the Problems view,
 - Minimize the Problems view,
 - Switch to the Output view, and
 - Check whether the editor fits the container.

Before:
screencast 2020-08-12 13-22-37

After:
screencast 2020-08-12 13-42-23

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/output/src/browser/output-widget.tsx [hunk 1]
```diff
@@ -147,6 +147,11 @@ export class OutputWidget extends BaseWidget implements StatefulWidget {
         }
     }
 
+    protected onAfterShow(msg: Message): void {
+        super.onAfterShow(msg);
+        this.onResize(Widget.ResizeMessage.UnknownSize); // Triggers an editor widget resize. (#8361)
```
Comment (akosyakov) [inline packages/output/src/browser/output-widget.tsx:152]: I wonder what is the difference to explicit `this.editor.refresh();`.

In `EditorWidget` we do it, we do it as well for `onAfterAttach` to resize it if a widget gets moved between areas.

Comment (kittaakos) [reply packages/output/src/browser/output-widget.tsx:152]: Yes, I started by calling `editor.refresh()` but it did not fix the problem.

Comment (kittaakos) [reply packages/output/src/browser/output-widget.tsx:152]: `this.autoSizing` is `false` in the monaco editor, so calling `refresh` is a NOOP.

Comment (kittaakos) [reply packages/output/src/browser/output-widget.tsx:152]: It turned out, calling `editor.refresh()` does nothing at all when invoked from `EditorWidget#onAfterShow`. `autoSizing` is always `false`.

Comment (kittaakos) [reply packages/output/src/browser/output-widget.tsx:152]: > In `EditorWidget` we do it, we do it as well for `onAfterAttach` to resize it if a widget gets moved between areas.

I have tried opening, maximizing, and DNDing editors while having breakpoints on both `onAfterAttach` and `onAfterShow` in the editor widget:

https://github.com/eclipse-theia/theia/blob/565a23c6d7e1ed763704897730a8760faf57dc09/packages/editor/src/browser/editor-widget.ts#L63

https://github.com/eclipse-theia/theia/blob/565a23c6d7e1ed763704897730a8760faf57dc09/packages/editor/src/browser/editor-widget.ts#L69

Neither of them has any effect as the underlying monaco editor's `autoresize` is a noop.

Comment (azatsarynnyy) [reply packages/output/src/browser/output-widget.tsx:152]: Maybe, somehow related to the old issue? https://github.com/microsoft/monaco-editor/issues/794

**Pull request merged**